### DISCUSSION TITLE
Add a blockstate property to a turtle (1.19.2)

### DISCUSSION THREAD

**Discussion by blockninja124** [cc-tweaked/CC-Tweaked#1832] [Q&A] [answered]
So now that I've got my peripheral on my turtle, I've run into another problem. Since my peripheral interacts with create mod, it has to pass a check where create checks whether it has a block state property called RAIL_SHAPE. I was wondering if its possible to add a blockstate to a turtle, or would I have to build my own turtle sub-class with the custom blockstate. I technically don't need to change the property, just have it exist, but that probably doesn't change anything on how I need to add it.

**Comment by SquidDev** [comment, marked as answer]
No, that's not possible, and I think this is probably a bad route to go down. It's probably worthwhile talking to the Create team to see if they can add a more flexible interface, rather than relying on `RAIL_SHAPE`.

**Reply by blockninja124** [reply]
I'll have a look on the Create github and see if I can get Create to change something. Thanks for all your help!

**Reply by blockninja124** [reply]
Actually turns out I was being kinda stupid with this one. 🤦I ended up simply creating a custom class that extended the create mod one and overode the function to get around the check.
